### PR TITLE
Caseflow: Adds degraded service override

### PR DESCRIPTION
**Pull request by evankroske**
## Summary
Adds an option to manually enable the degraded service banner. Fixes https://github.com/department-of-veterans-affairs/caseflow/issues/2703.

## Test Plan
When running Caseflow locally:
- [x] Banner appeared when told to
- [x] Banner disappeared when told to

### CODE REVIEW

**Comment by amprokop** [comment]
hey @evankroske, check out the linting failures above. you can use `rake lint` to just run the linter, and where it can, it will fix the code it deems unworthy.

File: README.md [hunk 1]
```diff
@@ -261,7 +261,19 @@ Rails.cache.write("reader_out_of_service", true)
 
 # to disable, e.g.
 Rails.cache.write("certification_out_of_service", false)
+```
 
+## Degraded Service
+To enable the "Degraded Service" banner:
+```
+rails c
```
Comment (amprokop) [inline README.md]: i think we can remove the `rails c` for consistency with the above examples, also you need to run other commands to get into the console in prod.

File: app/services/dependencies_report_service.rb [hunk 2]
```diff
@@ -2,15 +2,21 @@ class DependenciesReportService
   class << self
     # this method is in case we need list of dependencies/services that are degraded
     def find_degraded_dependencies
-      report = JSON.parse Rails.cache.read(:dependencies_report)
+      str_report = Rails.cache.read(:dependencies_report)
+      if not str_report
```
Comment (amprokop) [inline app/services/dependencies_report_service.rb]: 👍

File: app/services/dependencies_report_service.rb [hunk 3]
```diff
@@ -2,15 +2,21 @@ class DependenciesReportService
...
         result << element["name"] if element["up_rate_5"].to_i < 50
       end
     end
 
     def outage_present?
-      find_degraded_dependencies.present?
-    rescue
+      find_degraded_dependencies.present? or Rails.cache.read(:degraded_service)
+    rescue => error
+      Rails.logger.warn "Exception thrown while checking dependency "\
```
Comment (amprokop) [inline app/services/dependencies_report_service.rb]: 👍

File: README.md [hunk 4]
```diff
@@ -261,7 +261,19 @@ Rails.cache.write("reader_out_of_service", true)
 
 # to disable, e.g.
 Rails.cache.write("certification_out_of_service", false)
+```
 
+## Degraded Service
+To enable the "Degraded Service" banner:
```
Comment (amprokop) [inline README.md]: Can we add a bit more detail here just so it's clear when you would want to do this?

For example:

We show a "Degraded Service" banner across all Caseflow applications automatically when [Caseflow Monitor](https://github.com/department-of-veterans-affairs/caseflow-monitor) detects that our dependencies may be down. To enable this banner manually, overriding our automatic checks:

File: README.md [hunk 5]
```diff
@@ -261,7 +261,19 @@ Rails.cache.write("reader_out_of_service", true)
...
+```
+rails c
+Rails.cache.write(:degraded_service, true)
+```
+
+To disable the "Degraded Service" banner:
```
Comment (amprokop) [inline README.md]: It'd be great to be clear that we're not actually disabling the banner by setting this value to false (even with this PR, there is no way to do that so far)

To disable the manual override for the Degraded Service banner, letting [Caseflow Monitor](https://github.com/department-of-veterans-affairs/caseflow-monitor) again decide when this banner should appear:

File: README.md (hunk 4 above)
Comment (evankroske) [inline README.md]: Done

File: README.md (hunk 1 above)
Comment (evankroske) [inline README.md]: Done

**Comment by evankroske** [comment]
Fixed the lint errors

File: app/services/dependencies_report_service.rb [hunk 6]
```diff
@@ -2,15 +2,25 @@ class DependenciesReportService
...
+      return [] if !str_report
+      report = JSON.parse str_report
       report.values.each_with_object([]) do |element, result|
         result << element["name"] if element["up_rate_5"].to_i < 50
       end
     end
 
     def outage_present?
+      case Rails.cache.read(:degraded_service_banner)
+      when :always_show
```
Comment (amprokop) [inline app/services/dependencies_report_service.rb]: 👍

**Pull request merged**